pkg/trace/writer/stats.go

```diff
@@ -95,6 +109,29 @@ func (w *StatsWriter) Run() {
 	}
 }
 
+// FlushSync is a no-op for TraceWriter
+func (w *StatsWriter) FlushSync() error {
+	if !w.syncMode {
+		return errors.New("SyncFlush called on StatsWriter, which is a no-op")
+	}
+
+	defer w.report()
+	notify := make(chan struct{}, 1)
+	w.flushChan <- notify
+	<-notify
+
+	var wg sync.WaitGroup
+	for _, s := range w.senders {
+		wg.Add(1)
+		go func(s *sender) {
+			defer wg.Done()
+			s.waitForInflight()
+		}(s)
+	}
+	wg.Wait()
```

Comment:
This logic is out of place here. It feels like it should be in the select case of `flushChan`. Then the sending will be paired with the waiting.

The select statement will be uglier and bigger but that's fine.